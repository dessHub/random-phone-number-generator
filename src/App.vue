<template>
  <v-app id="app">
    <div id="action-buttons">
      <span>
        <Button v-bind:getNumbers="getNumbers" />
      </span>
    </div>
    <TableComponent msg="Welcome to Your Vue.js App"
                    v-bind:numbers="numbers"/>
  </v-app>
</template>

<script>
import TableComponent from './components/TableComponent.vue'
import Button from './components/Button.vue'
import { getRandomNumbers } from './helpers';

export default {
  name: 'app',
  components: {
   TableComponent,
   Button
  },
  created(){
    this.numbers = getRandomNumbers();
  },
  data() {
    return {
      numbers: [],
    }
  },
  methods: {
    getNumbers: function() {
      this.numbers = getRandomNumbers();
      return ;
    },
  }
}
</script>

<style>
#app {
  font-family: 'Avenir', Helvetica, Arial, sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  text-align: center;
  color: #2c3e50;
  margin-top: 60px;
}

#action-buttons {
  float: left;
}

#action-buttons span {
  display: inline-block;
}
</style>
