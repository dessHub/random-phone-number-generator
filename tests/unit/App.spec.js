import { mount } from '@vue/test-utils'
import App from '@/App.vue'

describe('App.vue', () => {
  it('renders correctly', () => {
  const wrapper = mount(App)
  expect(wrapper.element).toMatchSnapshot()
  });
  it('has a created hook', () => {
    expect(typeof App.created).toBe('function')
  });
  it('fetches new numbers when button clicked', () => {
    const wrapper = mount(App);
    const getNumbers = jest.fn();
    wrapper.setMethods({getNumbers});
    const button = wrapper.find('.btn-generate');
    button.trigger('click');
    expect(getNumbers).toBeCalled();
  })
})
