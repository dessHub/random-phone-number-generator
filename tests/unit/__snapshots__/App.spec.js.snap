// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.vue renders correctly 1`] = `
<div
  class="application theme--light"
  data-app="true"
  id="app"
>
  <div
    class="application--wrap"
  >
    <div
      class="layout"
    >
      <div
        class="flex xs12 sm10 offset-sm1"
      >
        <div
          class="v-card v-sheet theme--light"
        >
          <div
            class="container fill-height fluid"
          >
            <div
              class="layout fill-height"
            >
              <div
                class="flex xs12 align-end flexbox"
              >
                <span
                  class="headline"
                >
                  Random Phone Number Generator
                </span>
                 
                <div
                  class="v-card__title"
                >
                  <div>
                    <p>
                      1. Generate Phone Numbers by selecting Min, Max and
                            Quantity. Then hit generate button when it is
                            active.
                    </p>
                     
                    <p>
                      2. You can sort phone numbers
                     by clicking on the table column name i.e Phone
                     Numbers. Whe arrow up, items are sort by
                     ascending order.
                    </p>
                  </div>
                </div>
                 
                <form
                  class="v-form"
                  novalidate="novalidate"
                >
                  <div
                    class="container grid-list-xl fluid"
                  >
                    <div
                      class="layout wrap"
                    >
                      <div
                        class="flex xs12 sm3"
                      >
                        <div
                          class="v-input v-input--slider v-input--slider--thumb-label theme--light"
                          id="min"
                        >
                          <div
                            class="v-input__control"
                          >
                            <div
                              class="v-input__slot"
                            >
                              <label
                                class="v-label theme--light"
                                for="min"
                                style="left: 0px; position: relative;"
                              >
                                Min
                              </label>
                              <div
                                class="v-slider"
                              >
                                <input
                                  aria-label="Min"
                                  aria-readonly="false"
                                  aria-valuemax="99999998"
                                  aria-valuemin="10000000"
                                  id="min"
                                  readonly="readonly"
                                  role="slider"
                                />
                                <div
                                  class="v-slider__track__container"
                                >
                                  <div
                                    class="v-slider__track"
                                    style="left: 7px; width: 100%;"
                                  />
                                  <div
                                    class="v-slider__track-fill orange"
                                    style="left: 0px; width: 0%;"
                                  />
                                </div>
                                <div
                                  class="v-slider__thumb-container v-slider__thumb-container--show-label"
                                  style="left: 0%;"
                                >
                                  <div
                                    class="v-slider__thumb"
                                  />
                                  <div
                                    class="v-slider__thumb-label__container"
                                    style="display: none;"
                                  >
                                    <div
                                      class="v-slider__thumb-label"
                                      style="height: 32px; width: 32px;"
                                    >
                                      <span />
                                    </div>
                                  </div>
                                </div>
                              </div>
                            </div>
                            <div
                              class="v-messages theme--light"
                            >
                              <span
                                class="v-messages__wrapper"
                                name="message-transition"
                                tag="div"
                              />
                            </div>
                          </div>
                        </div>
                      </div>
                       
                      <div
                        class="flex xs12 sm3"
                      >
                        <div
                          class="v-input v-input--slider v-input--slider--thumb-label theme--light"
                          id="max"
                        >
                          <div
                            class="v-input__control"
                          >
                            <div
                              class="v-input__slot"
                            >
                              <label
                                class="v-label theme--light"
                                for="max"
                                style="left: 0px; position: relative;"
                              >
                                Max
                              </label>
                              <div
                                class="v-slider"
                              >
                                <input
                                  aria-label="Max"
                                  aria-readonly="false"
                                  aria-valuemax="99999999"
                                  aria-valuemin="10000001"
                                  id="max"
                                  readonly="readonly"
                                  role="slider"
                                />
                                <div
                                  class="v-slider__track__container"
                                >
                                  <div
                                    class="v-slider__track"
                                    style="left: 7px; width: 100%;"
                                  />
                                  <div
                                    class="v-slider__track-fill orange"
                                    style="left: 0px; width: 0%;"
                                  />
                                </div>
                                <div
                                  class="v-slider__thumb-container v-slider__thumb-container--show-label"
                                  style="left: 0%;"
                                >
                                  <div
                                    class="v-slider__thumb"
                                  />
                                  <div
                                    class="v-slider__thumb-label__container"
                                    style="display: none;"
                                  >
                                    <div
                                      class="v-slider__thumb-label"
                                      style="height: 32px; width: 32px;"
                                    >
                                      <span />
                                    </div>
                                  </div>
                                </div>
                              </div>
                            </div>
                            <div
                              class="v-messages theme--light"
                            >
                              <span
                                class="v-messages__wrapper"
                                name="message-transition"
                                tag="div"
                              />
                            </div>
                          </div>
                        </div>
                      </div>
                       
                      <div
                        class="flex xs12 sm4"
                      >
                        <div
                          class="v-input v-input--slider v-input--slider--thumb-label theme--light"
                          id="quantity"
                        >
                          <div
                            class="v-input__control"
                          >
                            <div
                              class="v-input__slot"
                            >
                              <label
                                class="v-label theme--light"
                                for="quantity"
                                style="left: 0px; position: relative;"
                              >
                                Quantity
                              </label>
                              <div
                                class="v-slider"
                              >
                                <input
                                  aria-label="Quantity"
                                  aria-readonly="false"
                                  aria-valuemax="10000"
                                  aria-valuemin="1"
                                  id="quantity"
                                  readonly="readonly"
                                  role="slider"
                                />
                                <div
                                  class="v-slider__track__container"
                                >
                                  <div
                                    class="v-slider__track"
                                    style="left: 7px; width: 100%;"
                                  />
                                  <div
                                    class="v-slider__track-fill orange"
                                    style="left: 0px; width: 0%;"
                                  />
                                </div>
                                <div
                                  class="v-slider__thumb-container v-slider__thumb-container--show-label"
                                  style="left: 0%;"
                                >
                                  <div
                                    class="v-slider__thumb"
                                  />
                                  <div
                                    class="v-slider__thumb-label__container"
                                    style="display: none;"
                                  >
                                    <div
                                      class="v-slider__thumb-label"
                                      style="height: 32px; width: 32px;"
                                    >
                                      <span />
                                    </div>
                                  </div>
                                </div>
                              </div>
                            </div>
                            <div
                              class="v-messages theme--light"
                            >
                              <span
                                class="v-messages__wrapper"
                                name="message-transition"
                                tag="div"
                              />
                            </div>
                          </div>
                        </div>
                      </div>
                       
                      <div
                        class="flex xs12 sm1"
                      >
                        <button
                          class="btn-generate v-btn v-btn--disabled v-btn--round theme--light normal--text"
                          disabled="disabled"
                          type="submit"
                        >
                          <div
                            class="v-btn__content"
                          >
                            Generate
                          </div>
                        </button>
                      </div>
                    </div>
                     
                    <div
                      class="layout wrap"
                    >
                      <div
                        class="flex xs12 sm3"
                      >
                        <span>
                          Min - 0710000000
                        </span>
                      </div>
                       
                      <div
                        class="flex xs12 sm3"
                      >
                        <span>
                          Max - 0799999999
                        </span>
                      </div>
                       
                      <div
                        class="flex xs12 sm3"
                      >
                        <span>
                          10 Phone Numbers
                        </span>
                      </div>
                       
                      <div
                        class="flex xs12 sm3"
                      >
                        <download-csv
                          data="[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]"
                          name="filename.csv"
                        >
                          <button
                            class="v-btn v-btn--round theme--dark normal"
                            type="button"
                          >
                            <div
                              class="v-btn__content"
                            >
                              Download CSV
                            </div>
                          </button>
                        </download-csv>
                      </div>
                    </div>
                  </div>
                </form>
              </div>
            </div>
          </div>
           
          <div
            class="container"
          >
            <div>
              <div
                class="elevation-1"
              >
                <div
                  class="v-table__overflow"
                >
                  <table
                    class="v-datatable v-table theme--light"
                  >
                    <thead>
                      <tr>
                        <th
                          aria-label="Index: Not sorted. Activate to sort ascending."
                          aria-sort="none"
                          class="column sortable text-xs-left"
                          role="columnheader"
                          scope="col"
                          tabindex="0"
                        >
                          Index
                          <i
                            aria-hidden="true"
                            class="v-icon material-icons theme--light"
                            style="font-size: 16px;"
                          >
                            arrow_upward
                          </i>
                        </th>
                        <th
                          aria-label="Phone Numbers: Not sorted. Activate to sort ascending."
                          aria-sort="none"
                          class="column sortable text-xs-left"
                          role="columnheader"
                          scope="col"
                          tabindex="0"
                        >
                          Phone Numbers
                          <i
                            aria-hidden="true"
                            class="v-icon material-icons theme--light"
                            style="font-size: 16px;"
                          >
                            arrow_upward
                          </i>
                        </th>
                      </tr>
                      <tr
                        class="v-datatable__progress"
                      >
                        <th
                          class="column"
                          colspan="2"
                        />
                      </tr>
                    </thead>
                    <tbody>
                      <tr>
                        <td
                          class="text-xs-center"
                          colspan="2"
                        >
                          No matching records found
                        </td>
                      </tr>
                    </tbody>
                  </table>
                </div>
                <div
                  class="v-datatable v-table theme--light"
                >
                  <div
                    class="v-datatable__actions"
                  >
                    <div
                      class="v-datatable__actions__select"
                    >
                      Rows per page:
                      <div
                        class="v-input v-text-field v-select v-input--hide-details v-input--is-label-active v-input--is-dirty theme--light"
                        role="combobox"
                      >
                        <div
                          class="v-input__control"
                        >
                          <div
                            class="v-input__slot"
                          >
                            <div
                              class="v-select__slot"
                            >
                              <div
                                class="v-select__selections"
                              >
                                <div
                                  class="v-select__selection v-select__selection--comma"
                                >
                                  25
                                </div>
                                <input
                                  aria-label="Rows per page:"
                                  aria-readonly="false"
                                  autocomplete="on"
                                  readonly="readonly"
                                  type="text"
                                />
                              </div>
                              <div
                                class="v-input__append-inner"
                              >
                                <div
                                  class="v-input__icon v-input__icon--append"
                                >
                                  <i
                                    aria-hidden="true"
                                    class="v-icon material-icons theme--light"
                                  >
                                    arrow_drop_down
                                  </i>
                                </div>
                              </div>
                            </div>
                            <div
                              class="v-menu"
                            >
                              <div
                                class="v-menu__content theme--light v-menu__content--auto "
                                style="max-height: 200px; min-width: 75px; max-width: auto; top: 12px; left: 0px; transform-origin: top left; z-index: 0; display: none;"
                              >
                                <div
                                  class="v-select-list v-card theme--light"
                                >
                                  <div
                                    class="v-list theme--light"
                                    role="list"
                                  >
                                    <div
                                      role="listitem"
                                    >
                                      <a
                                        class="v-list__tile v-list__tile--link theme--light"
                                      >
                                        <div
                                          class="v-list__tile__content"
                                        >
                                          <div
                                            class="v-list__tile__title"
                                          >
                                            5
                                          </div>
                                        </div>
                                      </a>
                                    </div>
                                    <div
                                      role="listitem"
                                    >
                                      <a
                                        class="v-list__tile v-list__tile--link theme--light"
                                      >
                                        <div
                                          class="v-list__tile__content"
                                        >
                                          <div
                                            class="v-list__tile__title"
                                          >
                                            10
                                          </div>
                                        </div>
                                      </a>
                                    </div>
                                    <div
                                      class="primary--text"
                                      role="listitem"
                                    >
                                      <a
                                        class="v-list__tile v-list__tile--link v-list__tile--active theme--light primary--text"
                                      >
                                        <div
                                          class="v-list__tile__content"
                                        >
                                          <div
                                            class="v-list__tile__title"
                                          >
                                            25
                                          </div>
                                        </div>
                                      </a>
                                    </div>
                                    <div
                                      role="listitem"
                                    >
                                      <a
                                        class="v-list__tile v-list__tile--link theme--light"
                                      >
                                        <div
                                          class="v-list__tile__content"
                                        >
                                          <div
                                            class="v-list__tile__title"
                                          >
                                            All
                                          </div>
                                        </div>
                                      </a>
                                    </div>
                                  </div>
                                </div>
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                    <div
                      class="v-datatable__actions__range-controls"
                    >
                      <div
                        class="v-datatable__actions__pagination"
                      >
                        NaN-NaN of 10
                      </div>
                      <button
                        aria-label="Previous page"
                        class="v-btn v-btn--flat v-btn--icon theme--light"
                        type="button"
                      >
                        <div
                          class="v-btn__content"
                        >
                          <i
                            aria-hidden="true"
                            class="v-icon material-icons theme--light"
                          >
                            chevron_left
                          </i>
                        </div>
                      </button>
                      <button
                        aria-label="Next page"
                        class="v-btn v-btn--flat v-btn--icon theme--light"
                        type="button"
                      >
                        <div
                          class="v-btn__content"
                        >
                          <i
                            aria-hidden="true"
                            class="v-icon material-icons theme--light"
                          >
                            chevron_right
                          </i>
                        </div>
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
